plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -201,8 +201,12 @@ public void initModel(final Connection connection, final IProject project, boole
 				protected IStatus doRun(IProgressMonitor monitor) {
 					try {
 						List<IProject> projects = connection.getResources(ResourceKind.PROJECT);
+						//Project might be set by a master model that set connection to this slave model
+						//and thus caused loading projects job. While this job was in wait,
+						//master model might set project. It must be respected.
```

Comment:
the structural difference I see in the other models and this one is that the other ones have the resources loaded once all the settings are done (they dont load upon #setConnection). While the deploy image model loads the resources implicitly when the connection is set. And this is, I think, what needs to be changed.